doc/content/xapi/migration_overview.md

```diff
@@ -0,0 +1,87 @@
+context, either to be forwared to a host or executed locally. Broadly, the
+context follows RBAC rules. The executed function is related to the message of
+the request (refer to [XenAPI Reference](https://xapi-project.github.io/xen-api/)).
+- In the case of the migration you can refer to *ocaml/idl/datamodel_vm.ml*.
+- The server will dispatch the operation to server helpers, executing the
+operation synchronously or asynchronously and returning the RPC answer.
+- Operations can then be forwarded to another host in the pool or executed
+locally.
+- When executed locally the high-level migration operation is send to the
+*Xenopsd daemon* by posting a message on the message switch.
+*Xenopsd daemon* will split this operation into several *atomic* operations
+that will be run by the *xenopsd backend* in XAPI. The steps involve:
+    - Splitting operations into multiple backend operations.
+    - Posting this operation as an RPC to *xenopsd server*.
+    - The *xenopsd server* then pushes operations onto the message switch.
+    - This allows the *xenopsd backend* to retrieve them and execute them.
+    - interactions between the xenopsd server and its backend occur
+    bidirectionnaly using the broker.
+
+## A diagram is worth a thousand words
+
+```mermaid
+
+flowchart TB
+
+    %% First we are starting by a XAPI client that is sending an XML-RPC request
+    client((Xapi client)) --|sends RPC XML request|-->
+        xapi_server{"`Dispatch RPC
+                    **api_server.ml**`"}
+    style client stroke:#CAFEEE,stroke-width:4px
+
+    %% XAPI Toolstack internals
+    subgraph "Xapi Toolstack"
+        style server stroke:#BAFA00,stroke-width:4px,stroke-dasharray: 5 5
+
+        xapi_server --|dispatch call (ie VM.pool_migrate)|--> server("`Auto generated using *IDL*
+                **server.ml**`")
+
+        server --|do_dispatch (ie VM.pool_migrate)|--> server_helpers["`server helpers
+        **server_helpers.ml**`"]
+
+        server_helpers --| call management (ie xapi_vm_migrate.ml)|--> vm_management["`VM Mgmt
+        like **xapi_vm_migrate.ml**`"]
+
+        vm_management --| Call |-->
+        xapi_xenops["`Transform xenops
+        see (**xapi_xenpos.ml**)`"] --| Post following IDL model (see xenops_interface.ml |--> xenopsd_broker
+
+        subgraph "xenopsd"
+            xenopsd_server["`Xenposd Daemon
+            split high level opertion into atomics`"] <--|Push/Pop on org.xen.xapi.xenopsd.classic|-->
+                xenopsd_broker[/Message switch: Queues\]
+            xenopsd_server --| split call |-->  xenopsd_backend
+        end
+    end
+
+    %% Xenopsd backend is accessing xen and xenstore
+    xenopsd_backend["Xenopsd Backend\nXC (libxenctrl)"] --|access to|--> xen_hypervisor["Xen hypervisor & xenstore"]
+    style xen_hypervisor stroke:#BEEF00,stroke-width:2px
+
+    %% Can send request to master
+    vm_management --|forward|--> master["Xapi master host"]
+    style master stroke:#B0A,stroke-width:4px
```

Comment:
It's the other way around: the top part of the diagram always runs on the coordinator (master) host. Message_forwarding may send it to another host.

Reply:
Oh. You mean that if as a user I run a `xe` command on a slave it first sends the command to the master, execute the top part of the diagram on the master and eventually (if it needs to be run on the slave) the message forwarding will send it back to the slave.

Reply:
Yes exactly.